src/sentry/backup/imports.py

```diff
@@ -239,26 +245,36 @@ def _import(
 
         filters.append(email_filter)
 
-    # The input JSON blob should already be ordered by model kind. We simply break up 1 JSON blob
-    # with N model kinds into N json blobs with 1 model kind each.
-    def yield_json_models(content) -> Iterator[tuple[NormalizedModelName, str]]:
+    # The input JSON blob should already be ordered by model kind. We simply break it up into
+    # smaller chunks, while guaranteeing that each chunk contains at most 1 model kind.
+    def yield_json_models(content) -> Iterator[tuple[NormalizedModelName, str, int]]:
         # TODO(getsentry#team-ospo/190): Better error handling for unparsable JSON.
         models = json.loads(content)
         last_seen_model_name: NormalizedModelName | None = None
         batch: list[type[Model]] = []
+        num_current_model_instances_yielded = 0
         for model in models:
             model_name = NormalizedModelName(model["model"])
             if last_seen_model_name != model_name:
                 if last_seen_model_name is not None and len(batch) > 0:
-                    yield (last_seen_model_name, json.dumps(batch))
+                    yield (
+                        last_seen_model_name,
+                        json.dumps(batch),
+                        num_current_model_instances_yielded,
+                    )
 
+                num_current_model_instances_yielded = 0
                 batch = []
                 last_seen_model_name = model_name
+            if len(batch) >= MAX_BATCH_SIZE:
+                yield (last_seen_model_name, json.dumps(batch), num_current_model_instances_yielded)
```

Comment:
Should `num_current_model_instances_yielded` include the count from the current batch? Or is the intent more to measure the number of records that were yielded before the current batch?

Reply:
The idea is to set an offset for the current batch. It answers the question: "how many models of type M have we already yielded not counting the ones in this batch." This allows us to offset the ordinals for this batch accordingly as we ship it over the wire.

I've adjusted the language in the comment above this to include this.